### ISSUE TITLE
@hits contains links that point to missing features

### ISSUE THREAD

**Issue by lgatto** [rformassspectrometry/QFeatures#141]
From @statomics 
```r
pe <- normalize(pe, 
                i = "peptideLog", 
                name = "peptideNorm", 
                method = "center.median”)
```
produces
```
Error in .checkLinksInHits(al@hits, object[[al@name]], direction = "to") : @hits contains links that point to missing features
15. stop("@hits contains links that point ", direction, " missing features")
14. .checkLinksInHits(al@hits, object[[al@name]], direction = "to")
13. .validAssayLink(object, i)
12. .validAssayLinks(object)
11. validityMethod(object)
10. isTRUE(x)
9. anyStrings(validityMethod(object))
8. validObject(.Object)
7. initialize(value, ...)
6. initialize(value, ...)
5. new("QFeatures", ExperimentList = el, colData = colData(x), sampleMap = smap, metadata = metadata(x), assayLinks = append(x@assayLinks, assayLinks))
4. addAssay(object, normalize(object[[i]], method, ...), name)
3. .local(object, ...)
2. normalize(pe, i = "peptideLog", name = "peptideNorm", method = "center.median")
1. normalize(pe, i = "peptideLog", name = "peptideNorm", method = "center.median")
```

This happens with the latest github version, not with the one from Bioconductor. 

```SESSION INFO

R version 4.1.0 Patched (2021-05-23 r80364)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Big Sur 11.2

Matrix products: default
LAPACK: /Library/Frameworks/R.framework/Versions/4.1-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] parallel  stats4    stats     graphics 
[5] grDevices utils     datasets  methods  
[9] base     

other attached packages:
 [1] gridExtra_2.3              
 [2] plotly_4.9.4.1             
 [3] forcats_0.5.1              
 [4] stringr_1.4.0              
 [5] dplyr_1.0.7                
 [6] purrr_0.3.4                
 [7] readr_1.4.0                
 [8] tidyr_1.1.3                
 [9] tibble_3.1.2               
[10] tidyverse_1.3.1            
[11] msqrob2gui_0.0.1           
[12] openxlsx_4.2.4             
[13] ggplot2_3.3.5              
[14] limma_3.48.1               
[15] ExploreModelMatrix_1.4.0   
[16] shinybusy_0.2.2            
[17] DT_0.18                    
[18] shinythemes_1.2.0          
[19] shinyjs_2.0.0              
[20] shiny_1.6.0                
[21] msqrob2_1.0.0              
[22] QFeatures_1.3.4            
[23] MultiAssayExperiment_1.18.0
[24] SummarizedExperiment_1.22.0
[25] Biobase_2.52.0             
[26] GenomicRanges_1.44.0       
[27] GenomeInfoDb_1.28.1        
[28] IRanges_2.26.0             
[29] S4Vectors_0.30.0           
[30] BiocGenerics_0.38.0        
[31] MatrixGenerics_1.4.0       
[32] matrixStats_0.59.0         

loaded via a namespace (and not attached):
 [1] minqa_1.2.4            
 [2] colorspace_2.0-2       
 [3] ellipsis_0.3.2         
 [4] XVector_0.32.0         
 [5] fs_1.5.0               
 [6] rstudioapi_0.13        
 [7] clue_0.3-59            
 [8] fansi_0.5.0            
 [9] lubridate_1.7.10       
[10] xml2_1.3.2             
[11] codetools_0.2-18       
[12] splines_4.1.0          
[13] cachem_1.0.5           
[14] knitr_1.33             
[15] jsonlite_1.7.2         
[16] nloptr_1.2.2.2         
[17] broom_0.7.8            
[18] cluster_2.1.2          
[19] dbplyr_2.1.1           
[20] shinydashboard_0.7.1   
[21] compiler_4.1.0         
[22] httr_1.4.2             
[23] backports_1.2.1        
[24] assertthat_0.2.1       
[25] Matrix_1.3-4           
[26] fastmap_1.1.0          
[27] lazyeval_0.2.2         
[28] cli_3.0.0              
[29] later_1.2.0            
[30] htmltools_0.5.1.1      
[31] tools_4.1.0            
[32] gtable_0.3.0           
[33] glue_1.4.2             
[34] GenomeInfoDbData_1.2.6 
[35] Rcpp_1.0.6             
[36] cellranger_1.1.0       
[37] jquerylib_0.1.4        
[38] vctrs_0.3.8            
[39] nlme_3.1-152           
[40] rintrojs_0.3.0         
[41] xfun_0.24              
[42] rvest_1.0.0            
[43] lme4_1.1-27.1          
[44] mime_0.11              
[45] lifecycle_1.0.0        
[46] zlibbioc_1.38.0        
[47] MASS_7.3-54            
[48] scales_1.1.1           
[49] hms_1.1.0              
[50] promises_1.2.0.1       
[51] ProtGenerics_1.24.0    
[52] AnnotationFilter_1.16.0
[53] RColorBrewer_1.1-2     
[54] yaml_2.2.1             
[55] sass_0.4.0             
[56] stringi_1.6.2          
[57] boot_1.3-28            
[58] zip_2.2.0              
[59] BiocParallel_1.26.0    
[60] rlang_0.4.11.9000      
[61] pkgconfig_2.0.3        
[62] bitops_1.0-7           
[63] evaluate_0.14          
[64] lattice_0.20-44        
[65] htmlwidgets_1.5.3      
[66] cowplot_1.1.1          
[67] tidyselect_1.1.1       
[68] magrittr_2.0.1         
[69] R6_2.5.0               
[70] generics_0.1.0         
[71] DelayedArray_0.18.0    
[72] DBI_1.1.1              
[73] pillar_1.6.1           
[74] haven_2.4.1            
[75] withr_2.4.2            
[76] MsCoreUtils_1.4.0      
[77] RCurl_1.98-1.3         
[78] modelr_0.1.8           
[79] crayon_1.4.1           
[80] utf8_1.2.1             
[81] rmarkdown_2.9          
[82] grid_4.1.0             
[83] readxl_1.3.1           
[84] data.table_1.14.0      
[85] reprex_2.0.0           
[86] digest_0.6.27          
[87] xtable_1.8-4           
[88] httpuv_1.6.1           
[89] munsell_0.5.0          
[90] viridisLite_0.4.0      
[91] bslib_0.2.5.1
```

@cvanderaa - any idea? Could this be related the latest modification with the assay link validity?

**Comment by cvanderaa** [comment]
Yes it is clearly related to our latest PR to `QFeatures`. However, `normalize` should not be able (by design) to generate this error... Do you know if there is a reproducible example available? I suspect that the `pe` object is not valid from the start, this can be tested by running `validObject(pe)`.

**Comment by lgatto** [comment]
@statOmics - don't hesitate to follow up here.

**Comment by cvanderaa** [comment]
`filterNA` currently creates invalid `QFeatures` objects. I will fix this bug. @statOmics could you confirm that `pe` was processed through `filterNA` before running `normalize` as stated above?

**Comment by lgatto** [comment]
Fixed in PR #142
